### PR TITLE
[Orders] properly account for base and quote fees

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/exchanges/connectors/ccxt/ccxt_websocket_connector.py [hunk 1]
```diff
@@ -846,6 +846,7 @@ async def candle(self, candles: list, symbol=None, timeframe=None, **kwargs):
         :param timeframe: the feed timeframe
         :param kwargs: the feed kwargs
         """
+        self.logger.debug(f'candle: {symbol} {timeframe} {candles}')
```
Comment (Herklos) [inline octobot_trading/exchanges/connectors/ccxt/ccxt_websocket_connector.py:849]: Should we keep it ?

File: octobot_trading/personal_data/portfolios/types/spot_portfolio.py [hunk 2]
```diff
@@ -28,20 +29,23 @@ def update_portfolio_data_from_order(self, order):
...
-        # update market
+        # update quote
         if order.side == enums.TradeOrderSide.BUY:
-            new_quantity = -(order.filled_quantity * order.filled_price)
+            new_quantity = -(order.filled_quantity * order.filled_price) - quote_fees
             self._update_portfolio_data(order.market, total_value=new_quantity)
         else:
-            new_quantity = (order.filled_quantity * order.filled_price) - order.get_total_fees(order.market)
+            new_quantity = (order.filled_quantity * order.filled_price) - quote_fees
             self._update_portfolio_data(order.market, total_value=new_quantity, available_value=new_quantity)
```
[the remark this answers is not in the document]
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/portfolios/types/spot_portfolio.py:49]: Right, before we did not take into account fees in USDT when buying BTC/USDT or fees in BTC when selling BTC/USDT.
Now if an exchange is applying this kind of fee, we also handle it properly in simulator

[the remark this answers is not in the document]
Comment (Herklos) [reply octobot_trading/personal_data/portfolios/types/spot_portfolio.py:49]: Ok I see so the newly added fees can be = 0 depending on how exchange apply fees on the symbol

[the remark this answers is not in the document]
Comment (Herklos) [reply octobot_trading/personal_data/portfolios/types/spot_portfolio.py:49]: Ok I see so the newly added fees can be = 0 depending on how exchange apply fees on the symbol

[the remark this answers is not in the document]
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/portfolios/types/spot_portfolio.py:49]: There is a 0 and a value in always base or quote fees, right. Depending on the exchange, this changes

File: octobot_trading/exchanges/connectors/ccxt/ccxt_websocket_connector.py (hunk 1 above)
Comment (GuillaumeDSM) [reply octobot_trading/exchanges/connectors/ccxt/ccxt_websocket_connector.py]: 👍

**Review by Herklos — APPROVED** [review]
:+1:

**Pull request merged**